crates/ruff_python_semantic/src/model.rs

```diff
@@ -271,7 +271,16 @@ impl<'a> SemanticModel<'a> {
     /// module, e.g. `import builtins; builtins.open`. It *only* includes the bindings
     /// that are pre-populated in Python's global scope before any imports have taken place.
     pub fn has_builtin_binding(&self, member: &str) -> bool {
-        self.lookup_symbol(member)
+        self.has_builtin_binding_in_scope(member, self.scope_id)
+    }
+
+    /// Return `true` if `member` is bound as a builtin *in the scope we are currently visiting*.
```

Comment:
```suggestion
    /// Return `true` if `member` is bound as a builtin *in a given scope*.
```

And you need ot change the sentence on line 268 to add `*in the scope we are currently visiting*` to the end of it.